app/components/properties-panel/properties-panel.component.ts

```diff
@@ -3,32 +3,64 @@
 // Date created: November 26, 2016
 
 
-import { Component } from "@angular/core";
-import { SinapType } from "../../models/types";
-
+import { Component, SimpleChanges, Output, EventEmitter } from "@angular/core";
+import * as Type from "../../models/types";
 
 @Component({
     selector: "sinap-properties-panel",
     templateUrl: "./properties-panel.component.html",
-    styleUrls: ["../../styles/side-panel.component.css", "./properties-panel.component.css"]
+    styleUrls: ["../../styles/side-panel.component.css", "./properties-panel.component.css"],
 })
-
 export class PropertiesPanelComponent {
     public selectedEntity: PropertiedEntity | null = null;
     public console = console;
-    // get selectedEntity() {
-    //   return this._selectedEntity;
-    // }
-    // set selectedEntity(e) {
-    //   console.log(e);
-    //   this._selectedEntity = e;
-    // }
+
+    @Output() propertyChanged = new EventEmitter();
+
+    // TODO: 
+    // discuss with Daniel
+    // I want to wrap up this functionality into bind.directive
+    // so that we can just do:
+    // <input [snpBind]="selectedEntity" [group]="..." [key]="..." [keyPath]="...">
+    // I've mostly got it, but it needs to somehow link in the forms module
+    // I'll talk more in person.
```

Comment:
Yeah @Sheyne this is something we might want to sit down together and figure out. This area of angular changed considerable from angular 1.x so I'm as lost as you are as to the best way to accomplish this in angular 2.x. 
(Here's a resource explaining some of the changes: https://blog.thoughtram.io/angular/2016/10/13/two-way-data-binding-in-angular-2.html)